src/renderer/base/renderer.cpp

```diff
@@ -582,11 +582,11 @@ void Renderer::_PaintBufferOutput(_In_ IRenderEngine* const pEngine)
             // This means that we need 14,27 out of the backing buffer to fill in the 1,1 cell of the screen.
             const auto screenLine = Viewport::Offset(bufferLine, -view.Origin());
 
-            // Retrieve the cell information iterator limited to just this line we want to redraw.
-            auto it = buffer.GetCellDataAt(bufferLine.Origin(), bufferLine);
+            //auto it = buffer.GetCellDataAt(bufferLine.Origin(), bufferLine);
+            //_PaintBufferOutputHelper(pEngine, it, screenLine.Origin());
 
-            // Ask the helper to paint through this specific line.
-            _PaintBufferOutputHelper(pEngine, it, screenLine.Origin());
+            const auto& rowData = buffer.GetRowByOffset(row);

```

Comment:
This method is supposed to handle all kinds of dirty rect refreshing. I kind of made it only working for rows, which is basically all that's used right now. Should make it better though.

Reply:
This is probably what ended up causing the duplicate conpty output - by just re-rendering the entire row at the time, that's causing the text to get duplicated on the Terminal side